content/lessons/chapter-3/pool-1.tsx

```diff
@@ -1,105 +1,186 @@
 'use client'
 
-import { useTranslations } from 'hooks'
+import { useTranslations, useSaveAndProceed } from 'hooks'
+import { useState, useEffect } from 'react'
+import { Button } from 'shared'
+import { sleep } from 'utils'
 import { ProfileWithHashPower } from 'types'
-import { Card, ContributionBar, HashFrequency, Text } from 'ui'
+import { Card, HashFrequency, Text, HashrateChallenge, NonceCounter } from 'ui'
 import Profile from 'ui/common/Profile'
+import clsx from 'clsx'
 
 export const metadata = {
   title: 'chapter_three.pool_one.title',
   image: '/assets/images/chapter-2-intro-1.jpg',
   key: 'CH3POL1',
 }
 
-const TOTAL_BLOCKS = 100
+export default function Pool1({ lang }) {
+  const t = useTranslations(lang)
+  const [transactionsConfirmed, setTransactionsConfirmed] = useState(0)
+  const [bitcoinMined, setBitcoinMined] = useState(0)
+  const [step, setStep] = useState(0)
+  const [blocks1, setBlocks1] = useState(0)
+  const [blocks2, setBlocks2] = useState(0)
```

Comment:
Let's rename theset to `protagonistsBlockAmount` and `antagonistsBlockAmount` or something similar, blocks1 and blocks2 is not very descriptive